lib/valkyrie/file_repository.rb

```diff
@@ -31,16 +31,20 @@ def upload(file:, model: nil)
         new_path = base_path.join(model.try(:id).to_s, file.original_filename)
         FileUtils.mkdir_p(new_path.parent)
         FileUtils.mv(file.tempfile.path, new_path)
-        File.new(id: Valkyrie::ID.new("diskrepository://#{new_path}"), repository: self)
+        find_by(id: Valkyrie::ID.new("diskrepository://#{new_path}"))
```

Comment:
The problem with adding new ID types is they have to be supported by the persisters as data types in the properties. I could see maybe a general ExternalID type, to separate from Valkyrie::ID, but...*shrug*